applications/nrf_desktop/src/modules/dvfs.c

```diff
@@ -226,11 +322,11 @@ static void dvfs_retry_work_handler(struct k_work *work)
 {
 	LOG_DBG("Retrying to change DVFS frequency.");
 
-	enum dvfs_frequency_setting required_freq = check_required_frequency();
+	uint32_t required_freq = check_required_frequency();
 
-	__ASSERT_NO_MSG(required_freq != current_freq);
-
-	set_dvfs_freq(required_freq);
+	if (required_freq != current_freq) {
```

Comment:
Unless I missed something the work should not be scheduled if `required_freq == current_freq`. Can we bring back the assert for better module state control?

Reply:
But in meanwhile we could change required frequency:

We change states to requiered_frequency x from frequency z -> it fails -> we schedule retry
We change states back to requiered_frequency z -> we don't set frequency because retry in progress
Retry work timeout finishes and we have same frequency as current frequency.

Reply:
But is there a case where work handler is executed and frequencies are the same? Aren't we cancelling the work one we spot that the frequencies are the same?